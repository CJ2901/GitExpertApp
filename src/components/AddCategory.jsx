import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('Spider-man PS4');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;
        
        // setCategories(categories => [inputValue, ...categories]);
        setInputValue('');
        onNewCategory( inputValue.trim());
    }

    return (
        <>
            {/* Es lo mismo que onSubmit={(event) => onSubmit(event)} */}
            <form onSubmit={ onSubmit }>
                <input
                    type="text"
                    placeholder="Buscar gifs"
                    value={inputValue}
                    onChange={onInputChange}
                />
            </form>
        </>
    )
}
