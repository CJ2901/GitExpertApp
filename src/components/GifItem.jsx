import PropTypes from 'prop-types';

export const GifItem = ({title, url, id}) => {
  
    return (
        <div className="card">
            <img src={url} alt={title} />
            <h3>{title}</h3>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    id: PropTypes.string,
}