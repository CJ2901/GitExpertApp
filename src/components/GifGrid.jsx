import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { PropTypes } from 'prop-types';

export const GifGrid = ({ category }) => {

  // Primera utilizaci√≥n de un custom hook
  const { images, isLoading } = useFetchGifs(category);


  return (
    <>
      <h3>{category}</h3>

      {
        isLoading && (<h2>Cargando...</h2>)
      }

      <div className="card-grid">
        {
          images.map((image) =>
            <GifItem
              key={image.id}
              {...image} //Esparcir image: Todas las propiedades que vengan de image las estoy esparciendo en GifItem
            />
          )
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}