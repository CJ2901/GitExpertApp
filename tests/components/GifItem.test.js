

import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('Pruebas en GifItem', () => { 
    
    const title = 'Spiderman';
    const url = 'https://spider-man.com/spidery.jpg'
    const id = "SlkjLyujkgYU";

    test('debe mostrar match con el snapshot', () => { 
        /*
            1. Añadir PropTypes ??? yarn add prop-types
                a. title Obligatorio
                b. url Obligatorio 
            
            2. Evaluar Snapshot
        */
        const {container} = render(<GifItem title={title} url={url} id={id} />)
        expect(container).toMatchSnapshot()
    });
    
    test('should mostrar la imagen con el URL y ALT indicado', () => { 
        render(<GifItem title={title} url={url} id={id} />);

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );

     });

     test('should mostrar el título en el componente', () => { 
        render(<GifItem title={title} url={url} id={id} />);
        expect( screen.getByText(title) ).toBeTruthy();
        
      });

 })