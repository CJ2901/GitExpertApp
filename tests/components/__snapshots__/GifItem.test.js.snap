// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem debe mostrar match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Spiderman"
      src="https://spider-man.com/spidery.jpg"
    />
    <h3>
      Spiderman
    </h3>
  </div>
</div>
`;
